### PR TITLE
Make the 'is_member' field of 'channel' objects optional

### PR DESCRIPTION
**Pull request by peddie**
This addresses #27.

### CODE REVIEW

**Comment by peddie** [comment]
I'll hold off on merging this in case you're unhappy with the defaulting behavior.

File: src/Web/Slack/Types/Channel.hs [hunk 1]
```diff
@@ -22,7 +23,7 @@ data Channel = Channel { _channelId         :: ChannelId
                        , _channelMembers    :: Maybe [UserId]
                        , _channelTopic      :: Maybe Topic
                        , _channelPurpose    :: Maybe Purpose
-                       , _channelIsMember   :: Bool
+                       , _channelIsMember   :: Maybe Bool
```
Comment (mpickering) [inline src/Web/Slack/Types/Channel.hs]: What does `Maybe Bool` mean? Is `Nothing == False`?

Comment (peddie) [inline src/Web/Slack/Types/Channel.hs]: No; `Nothing` means we don't know anything about it (the API didn't return this field).

Comment (mpickering) [inline src/Web/Slack/Types/Channel.hs]: I just foresee that any piece of code which uses this field will wrap it in a `fromMaybe False`.

**Comment by peddie** [comment]
OK, I can make it default to `False`.  I guess we don't often expect to see events wherein the default channel is created, and creation events seem to be the only ones missing this field.

**Comment by peddie** [comment]
I've updated this diff so that both fields default to `False`.

**Pull request merged**